### PR TITLE
flask:2024-09-03

### PR DESCRIPTION
**Pull request by iwmstjp**
Add Delete function

## Summary by Sourcery

Add a delete function to the JavaScript for table row removal and enhance server-side logic to handle empty data submissions and manage 'Delete' and 'Duplicate' columns in form data.

New Features:
- Introduce a delete function in the JavaScript to allow users to remove rows from the table.

Enhancements:
- Add handling for empty data lists in the server-side logic to prevent errors when no data is submitted.
- Modify the form data parsing to include placeholders for 'Delete' and 'Duplicate' columns.

### CODE REVIEW

**Comment by gpetho** [comment]
@iwmstjp Please review these suggestions. All seem plausible to me, so it might make sense to implement them. I'm  not sure whether the first comment is that important, but the last one looks like a valid comment in terms of the DRY principle, and the other two look sensible for the sake of maintainability. I leave the decision to you. None of these are urgent, but if you agree that these would be good to implement, make a note for yourself for later.

**Comment by iwmstjp** [comment]
I am going to review these suggestions today.

File: flask-version/metagenomongo/static/index.js [hunk 1]
```diff
@@ -1,5 +1,28 @@
-// window.addEventListener('load', function () {
-// 	document.querySelectorAll('table td').forEach(elm => {
-// 		elm.contentEditable = true;
-// 	});
-// });
\ No newline at end of file
+function deleteRow(button) {
```
[the remark this answers is not in the document]
Comment (iwmstjp) [reply flask-version/metagenomongo/static/index.js:1]: Simplify this part according to the suggestion

File: flask-version/metagenomongo/static/index.js [hunk 2]
```diff
@@ -1,5 +1,28 @@
...
+  var row = button.parentNode.parentNode;
+  row.parentNode.removeChild(row);
+}
+
+window.addEventListener('load', function () {
+	// index_with_table
+    document.querySelectorAll('input[name$="86"]').forEach(elm => {
```
[the remark this answers is not in the document]
Comment (iwmstjp) [reply flask-version/metagenomongo/static/index.js:8]: To implement this suggestion, I need to use if statemen depending on specific name attributes. Therefore I will not change this part.

File: flask-version/metagenomongo/app.py [hunk 3]
```diff
@@ -85,14 +85,13 @@ def check_user(user_name):
 def parse_form_data(form_data):
```
[the remark this answers is not in the document]
Comment (iwmstjp) [reply flask-version/metagenomongo/app.py:85]: I implemented this suggestion.

File: flask-version/metagenomongo/app.py [hunk 4]
```diff
@@ -130,6 +129,14 @@ def empty_check(last_data):
             return True
     return False
 
+def data_list_check(data_list, errors, data):
```
[the remark this answers is not in the document]
Comment (iwmstjp) [reply flask-version/metagenomongo/app.py:132]: I implemented this suggestion except save() becasue of difference of the process.

**Pull request merged**